src/shared/ethtool-util.c

```diff
@@ -219,16 +220,16 @@ int ethtool_get_link_info(int *ethtool_fd, const char *ifname,
 }
 
 int ethtool_get_permanent_macaddr(int *ethtool_fd, const char *ifname, struct ether_addr *ret) {
+        _cleanup_close_ int fd = -1;
         _cleanup_free_ struct ethtool_perm_addr *epaddr = NULL;
         struct ifreq ifr;
         int r;
 
-        assert(ethtool_fd);
         assert(ifname);
         assert(ret);
 
-        if (*ethtool_fd < 0) {
-                r = ethtool_connect_or_warn(ethtool_fd, false);
+        if (!ethtool_fd || *ethtool_fd < 0) {
+                r = ethtool_connect_or_warn(ethtool_fd ?: &fd, false);
```

Comment:
instead of adding this ternary check everywhere, wouldn't it be nicer to add this line to the top once?

```c
if (!ethtool_fd)
        ethtool_fd = &fd;
```

That should be less prone to accidentally forgetting to do the full ternary op thing, because you can just always use `*ethtool_fd`, regardles sin which context

Reply:
Yeah, that's nicer.